### PR TITLE
Robustify AMT

### PR DESCRIPTION
**Pull request by Stebalien**
* Remove floats and `math.Pow`.
* Set and enforce a maximum height.
* Set the max index to `2^63-1` (see the code for the reasoning).

We still need to do things like (future work):

* Enforce a maximum count.
* Fail if the tree is deeper than it should be.

### CODE REVIEW

**Comment by Stebalien** [comment]
cc @whyrusleeping

File: amt.go [hunk 1]
```diff
@@ -458,14 +461,13 @@ func (n *Node) loadNode(ctx context.Context, bs cbor.IpldStore, i uint64, create
...
-func nodesForHeight(width, height int) uint64 {
-	val := math.Pow(float64(width), float64(height))
-	if val >= float64(math.MaxUint64) {
-		log.Errorf("nodesForHeight overflow! This should never happen, please report this if you see this log message")
-		return math.MaxUint64
+func nodesForHeight(height int) uint64 {
+	heightLogTwo := uint64(widthBits * height)
+	if heightLogTwo >= 64 {
+		// Should never happen. Max height is checked at all entry points.
+		panic("height overflow")
```
Comment (Stebalien) [inline amt.go:468]: I mean, I think panicing is better than logging and returning the _wrong_ thing, right?

**Comment by Stebalien** [comment]
I think this is wrong.

**Comment by Stebalien** [comment]
No, it's right. I've made the max index less arbitrary.

File: amt.go (hunk 1 above)
Comment (whyrusleeping) [reply amt.go:476]: probably, yes

**Comment by Stebalien** [comment]
@Kubuxu I don't have write access. Could you merge?

**Pull request merged**